src/components/form.RadioGroup/form.RadioGroup.tsx

```diff
@@ -0,0 +1,44 @@
+import { Form } from "../form";
+import { useForm } from 'react-hook-form';
+
+type FormData = {
+    diabetesType: string;
+};
+
+
+type DiabetesFormProps = {
+    title: string;
+    description: string;
+    options: string[];
+}
+
+const DiabetesForm = ({title, description, options}: DiabetesFormProps) => {
+    const {register, handleSubmit, formState: { errors }} = useForm<FormData>();
+
+    const onSubmit = (data: FormData) => {
+        if (!errors.diabetesType) {
+        alert(`You selected ${data.diabetesType}`)
+        } else {
+            alert( 'Please select a diabetes type');
+        }
+    };
+
+    return (
+        <Form onSubmit={handleSubmit(onSubmit)}>
+            <p>{title}</p>
+            <div {...register('diabetesType', {required: true})}>
+                {options.map((option, index) => (
```

Comment:
Remember this radiogroup needs three radio inputs, so confirm that this allows for that. I recommend setting up a storybook (looks like you don't have one) to test this out. You can reference the ```form``` directory to see how to do that, or reach out to me. I can help with that.